docs/game_data/spel2.lua

```diff
@@ -1235,6 +1235,39 @@ function set_frametime_unfocused(frametime) end
 ---Get engine target frametime when game is unfocused (1/framerate, default 1/33).
 ---@return double?
 function get_frametime_unfocused() end
+---Adds new custom type (group of ENT_TYPE) that can be later used in functions like get_entities_by or set_(pre/post)_entity_spawn
+---@param types ENT_TYPE[]
+---@return ENT_TYPE
+function add_custom_type(types) end
```

Comment:
So if this was to be used with the "clone entitydb and spoof id scenario", it would probably be nice if it could be called without arguments (should probably also add the usual variadic ENT_TYPEs) to simply reserve an unused id for a new type, that can then be nicely found with the get functions. Something like

```lua
reserved_id = add_custom_type()
custom_rock = EntityDB:new(ENT_TYPE.ITEM_ROCK)
custom_rock.id = reserved_id
some_rock.type = custom_rock
get_entities_by_type(reserved_id) -- some_rock
```